apps/voting/app/src/vote-hooks.js

```diff
@@ -0,0 +1,80 @@
+import { useState, useEffect } from 'react'
+import { useAragonApi } from '@aragon/api-react'
+
+// when `params` are updated, call `fn` and pass the result
+function usePromiseResult(fn, params, memoParams, defaultValue) {
+  const [result, setResult] = useState(defaultValue)
+  useEffect(() => {
+    let cancelled = false
+    fn(...params).then(value => {
```

Comment:
Perhaps we could avoid passing params, and instead just invoke a no-arg function? Adding an extra set of params is quite confusing, and makes it inevitable we'll ask ourselves which one is the called params and which one is the memo set.

Users can just use an arrow function or `.bind()` to make an argument-less function for this.

Reply:
Agreed, this hook has changed a little and it now lets users call the function themselves: https://github.com/aragon/aragon-apps/blob/voting-react-api/apps/voting/app/src/utils-hooks.js#L16-L32